File: 2020/01/c/01.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    FILE *fptr = fopen("challenge.txt", "r");

    int number;
    int numbers[200] = {0};

    for (int i = 0; i < 200; i++)
    {
        if (fscanf(fptr, "%d", &number) > 0)
        {
            numbers[i] = number;
        }
    }

    for (int i = 0; i < 200; i++)
    {
        for (int j = 0; j < 200; j++)
        {
            if (numbers[i] + numbers[j] == 2020)
            {
                printf("%d\n", numbers[i] * numbers[j]);
            }
        }
    }
}

File: 2020/01/js/01.js
const path = "challenge.txt";
const file = Bun.file(path);

const text = await file.text();
const numberList = text.split('\n').map(number => Number(number));

let toReturn = 0;

numberList.forEach(number1 => {
    numberList.forEach(number2 => {
        if (number1 + number2 === 2020) toReturn = number1 * number2
    })
});

console.log(toReturn);